morpho-checkpoint-node/tests/Test/Morpho/QSM.hs

```diff
@@ -335,46 +354,98 @@ prop_4 = noShrinking $
           majority = 2
         }
 
-setup :: Int -> Int -> IO [NodeHandle]
-setup nodesNum testId = do
+setup :: Int -> Int -> Int -> IO [NodeHandle]
+setup nodesNum requiredMajority testId = do
```

Comment:
I think this should just be of type `Config -> IO [NodeHandle]`. Extend `Config` to include all the necessary parameters.